app/pages/HomePage/HomePage.tsx

```diff
@@ -27,6 +32,79 @@ export interface HeroData {
   buttons: ButtonType[];
 }
 
+interface HomePageData {
+  heroSection: HeroData;
+  categoriesSection: FeaturedCategoriesData;
+  opportunitySection: OppEventCardData;
+  eventSection: OppEventCardData;
+  /* Fix naming in Sanity schema */
+  twoColumnContentSections: {
+    twoColumnContentSections: TwoColumnContent;
+  }[];
+}
+
+export const HomePage = () => {
+  const [homePageData, setHomePageData] = useState<HomePageData | null>(null);
+
+  useEffect(() => {
+    const fetchHomePageData = async () => {
+      const query = `*[_type == "homePage" && name == "Home Page"][0]{
+        'heroSection': *[_type == "hero" && name == "Home Hero"][0],
+        categoriesSection {...,'featuredCategoriesSection': featuredCategoriesSection[]->},
+        opportunitySection {...,'opportunityCards': opportunityCards[]->},
+        eventSection {...,'eventCards': eventCards[]->},
+        'twoColumnContentSections': *[_type == 'contentPageType' && name == 'Home']{
+      twoColumnContentSections[0]->
+    }
+      }`;
+
+      const result: HomePageData = await client.fetch(query);
+
+      setHomePageData(result);
+    };
+
+    fetchHomePageData();
+  }, []);
+
+  if (!homePageData) {
```

Comment:
Suggestion: do an exact `null` check here to be clear about runtime expectations matching the type. 
```suggestion
  if (homePageData === null) {
```